democracy/src/lib.rs

```diff
@@ -416,12 +433,14 @@ pub mod pallet {
 			let old_proposal_state = proposal.state;
 			let now = <pallet_timestamp::Pallet<T>>::get();
 			let proposal_action_identifier = proposal.action.clone().get_identifier();
-			let cancelled_at = Self::cancelled_at(proposal_action_identifier);
-			let proposal_cancelled =
-				cancelled_at.is_some() && proposal.start < cancelled_at.unwrap();
+			let last_approved_proposal_for_action =
+				Self::last_approved_proposal_for_action(proposal_action_identifier);
+			let proposal_cancelled_by_other = last_approved_proposal_for_action.is_some() &&
+				proposal.start < last_approved_proposal_for_action.unwrap().0;
 			let proposal_too_old = now - proposal.start > T::ProposalLifetime::get();
-			if proposal_cancelled || proposal_too_old {
-				proposal.state = ProposalState::Cancelled;
+			if proposal_cancelled_by_other {
+				proposal.state =
+					ProposalState::SupersededBy { id: last_approved_proposal_for_action.unwrap().1 }
 			} else {
 				// passing
 				if Self::is_passing(proposal_id)? {
```

Comment:
Complicated logic here, but it seems correct:

Passing:
1. Passing && longer than confirmation period -> approve
2. Passing && shorter than confirmation period && proposal too old -> reject
3. Passing && shorted than confirmation period && not too old -> confirming

Not confirming:
1. Not passing && too old -> reject
2. Not passing && confirming -> ongoing. Why do we check here for confirming? is this just a safety measure to not accidentally overwrite rejected/approved with ongoing?

Reply:
> Why do we check here for confirming? is this just a safety measure to not accidentally overwrite rejected/approved with ongoing?

No. we need to make sure a proposal falls out of confirming if the approval falls below the threshold

Reply:
Yes, I just mean that we probably want to set it to `ongoing` in any case, but we can only get to that branch of the if statement if it is in confirming anyhow, otherwise, we have a bug somewhere else in the code.